File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsJava/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicsjava/data/repository/AppRepository.java
package pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.repository;

import androidx.lifecycle.LiveData;

import java.util.List;

import javax.inject.Inject;

import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.Student;
import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.db.AppDao;
public class AppRepository {

    private final AppDao appDao;

    @Inject
    public AppRepository(AppDao appDao) {
        this.appDao = appDao;
    }

    public LiveData<List<Student>> readAllData() {
        return appDao.readAllData();
    }

    public void insert(Student student) {
        appDao.addItem(student);
    }
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicskotlin/data/repository/AppRepository.kt
package pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.repository

import androidx.lifecycle.LiveData
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.Student
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.db.AppDao
import javax.inject.Inject

class AppRepository @Inject constructor (private val appDao: AppDao) {
    val readAllData: LiveData<List<Student>> = appDao.readAllData()
    suspend fun insert(student: Student) = appDao.addItem(student)
}